### PR TITLE
Make flaky Stopwatch test more reliable

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/10107
cc: @mellinoe

### CODE REVIEW

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Release Build and Test please

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Release Build and Test please

**Comment by mellinoe** [comment]
LGTM, although it's unfortunate how some of these timing issues keep forcing us to use messy workarounds.

:+1:

File: src/System.Runtime.Extensions/tests/System/Diagnostics/Stopwatch.cs [hunk 1]
```diff
@@ -69,9 +69,17 @@ public static void StartNewAndRestart()
...
-            watch.Restart();
-            Assert.True(watch.IsRunning);
-            Assert.True(watch.Elapsed < elapsedSinceStart);
+            for (int attempt = 0; attempt < 5; attempt++)  // The comparison below could fail if we get very unlucky with when the thread gets preempted
+            {
+                watch.Restart();
+                Assert.True(watch.IsRunning);
+                try
+                {
+                    Assert.True(watch.Elapsed < elapsedSinceStart);
```
Comment (MichalStrehovsky) [inline src/System.Runtime.Extensions/tests/System/Diagnostics/Stopwatch.cs]: Does this test anything? If it fails on all 5 attempts we will just catch it and move along, no?

Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/Diagnostics/Stopwatch.cs]: Oops, no, I had a throw that i somehow managed to delete. Will fix.

Comment (mellinoe) [inline src/System.Runtime.Extensions/tests/System/Diagnostics/Stopwatch.cs]: Good catch! I was a bit quick on the merge, my mistake.

**Pull request merged**